File: psra/repository/src/main/java/by/arhor/psra/repository/model/Photo.java
package by.arhor.psra.repository.model;

import java.util.Set;

import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;
import org.springframework.data.mongodb.core.mapping.Field;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Document(collection = "photos")
public class Photo {
	
	@Id
	@Field("id")
	private String id;
	
	@Field("path")
    private String path;
	
	@Field("tags")
	private Set<Tag> tags;
    
}

File: psra/repository/src/main/scala/by/arhor/psra/repository/model/Photo.scala
package by.arhor.psra.repository.model

import by.arhor.psra.CoreVersion
import by.arhor.psra.repository.model.traits.Identifiable
import org.springframework.data.mongodb.core.mapping.{Document, Field}

import scala.beans.BeanProperty

object Photo {
  val serialVersionUID: Long = CoreVersion.SERIAL_VERSION_UID
}

@Document("photos")
class Photo extends Entity
               with Identifiable {

  @BeanProperty
  var path: String = _
  
  @BeanProperty
  var name: String = _
  
  @BeanProperty
  var description: String = _
  
  @BeanProperty
  var tags: Set[Tag] = _
  
  @BeanProperty
  var comments: List[Comment] = _
    
}
